src/azul/uuids.py

```diff
@@ -67,18 +71,28 @@ def validate_uuid_prefix(uuid_prefix: str) -> None:
     ...
     azul.uuids.InvalidUUIDPrefixError: '8F53' is not a valid UUID prefix.
 
-    >>> uuid = '8f53d355-b2fa-4bab-a2f2-6852d852d2ec'
-    >>> # noinspection PyUnresolvedReferences
-    >>> prefixes = [uuid[:n] for n in range(len(uuid))]
-    >>> for prefix in prefixes: validate_uuid_prefix(prefix)
+    >>> validate_uuid_prefix('8')
+
+    >>> validate_uuid_prefix('8f538f53')
+
+    >>> validate_uuid_prefix('8f538f5-')
+    Traceback (most recent call last):
+    ...
+    azul.RequirementError: Remove the trailing `-` from the uuid prefix: 8f538f5-
+
+    >>> validate_uuid_prefix('8f538f-')
+    Traceback (most recent call last):
+    ...
+    azul.RequirementError: Remove the trailing `-` from the uuid prefix: 8f538f-
 
-    >>> validate_uuid_prefix('aaaaaaaa-aaaa-')
-    >>> validate_uuid_prefix('aaaaaaaa-aaaa-a')
+    >>> validate_uuid_prefix('8f538f53a')
     Traceback (most recent call last):
     ...
-    azul.uuids.InvalidUUIDPrefixError: 'aaaaaaaa-aaaa-a' is not a valid UUID prefix.
+    azul.uuids.InvalidUUIDPrefixError: '8f538f53a' is not a valid UUID prefix.
     """
     valid_uuid_str = '26a8fccd-bbd2-4342-9c19-6ed7c9bb9278'
+    reject(uuid_prefix.endswith('-'),
+           f'Remove the trailing `-` from the uuid prefix: {uuid_prefix}')
```

Comment:
It's not a good idea to give specific remedies. Just describe what is wrong.

UUID is an acronym. LIke ID and URL.

Also, it looks like `8f538f53-a` would be considered valid. I thought we agreed to limit prefixes to the first dash-separated group in the hexadecimal representation of UUIDs because the third group has some fixed bits that can't be randomized.

Reply:
When validating UUID prefixes that start to specify values into the

Reply:
`8f538f53/0` should be considered valid. `8f538f53/1` should not. `8f538f5/1` should be considered valid. `8f538f5/2` should not. And so on. Any combination of common prefix and partition prefix length should be validated against that rule and invalid combinations should be rejected as early as possible, so ideally not so late as when the individual partitions are actually enumerated during a reindex.